koku/masu/external/accounts/hierarchy/aws/aws_org_unit_crawler.py

```diff
@@ -0,0 +1,147 @@
+            response = function(NextToken=response.get("NextToken"), **kwargs)
+            results = results + response[resource_key]
+        return results
+
+    def _crawl_accounts_per_id(self, ou, prefix):
+        """
+        List accounts for parents given an aws identifer.
+
+        Args:
+            parent_id: unique id for org unit you want to list.
+            prefix: The org unit tree path
+
+        See:
+        [1] https://docs.aws.amazon.com/cli/latest/reference/organizations/list-accounts-for-parent.html
+        [2] https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/organizations.html
+        """
+        parent_id = ou.get("Id")
+        LOG.info("Obtaining accounts for organizational unit: %s" % parent_id)
+        child_accounts = self._depaginate(
+            function=self.client.list_accounts_for_parent, resource_key="Accounts", ParentId=parent_id
+        )
+        for act_info in child_accounts:
+            self._save_aws_org_method(ou.get("Name", ou.get("Id")), ou.get("Id"), prefix, act_info.get("Id"))
+
+    def _crawl_org_for_acts(self, ou, prefix):
+        """
+        Recursively crawls the org units and accounts.
+
+        Args:
+            ou (dict): A return from aws client that includes the Id
+        """
+
+        # Save entry for current OU
+        self._save_aws_org_method(ou.get("Name", ou.get("Id")), ou.get("Id"), prefix, None)
+        try:
+
+            # process accounts for this org unit
+            self._crawl_accounts_per_id(ou, prefix)
+
+            # recurse and look for sub org units
+            ou_pager = self.client.get_paginator("list_organizational_units_for_parent")
+            for sub_ou in ou_pager.paginate(ParentId=ou.get("Id")).build_full_result().get("OrganizationalUnits"):
+                new_prefix = prefix + ("&%s" % sub_ou.get("Id"))
+                LOG.info("Organizational unit found during crawl: %s" % (sub_ou.get("Id")))
+                self._crawl_org_for_acts(sub_ou, new_prefix)
+        except Exception:
+            LOG.exception(
+                "Failure processing org unit.  Account schema {} and org_unit_id {}".format(self.schema, ou.get("Id"))
+            )
+
+    def _save_aws_org_method(self, unit_name, unit_id, unit_path, account_id=None):
+        """
+        Recursively crawls the org units and accounts.
+
+        Args:
+            ou (dict): A return from aws client that includes the Id
+        """
+        LOG.info(
+            "Saving account or org unit: unit_name=%s, unit_id=%s, unit_path=%s, account_id=%s"
+            % (unit_name, unit_id, unit_path, account_id)
+        )
+        with schema_context(self.schema):
+            row, created = AWSOrganizationalUnit.objects.get_or_create(
+                org_unit_name=unit_name, org_unit_id=unit_id, org_unit_path=unit_path, account_id=account_id
+            )
+            if created:
+                row.created_timestamp = DateAccessor().today()
```

Comment:
You can let Django handle this with a default value so you don't have to set it. (Totally optional, this is okay as-is)